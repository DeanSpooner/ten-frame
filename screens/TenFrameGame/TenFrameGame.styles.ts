import { StyleSheet } from "react-native";

const tenFrameGameStyles = StyleSheet.create({
  score: {
    color: "#ffe800",
    fontSize: 40,
    marginBottom: 50,
    textAlign: "center",
  },
  text: {
    color: "#fff",
    fontSize: 24,
    textAlign: "center",
    marginHorizontal: 15,
  },
  buttonsContainer: {
    flexWrap: "wrap",
    flexDirection: "row",
    maxWidth: 375,
    justifyContent: "center",
  },
  container: {
    backgroundColor: "#fff",
    borderColor: "#000",
    borderWidth: 3,
    borderRadius: 50,
    margin: 5,
    height: 65,
    width: 65,
  },
  buttonText: {
    fontSize: 48,
    textAlign: "center",
  },
  answerMessage: {
    fontSize: 36,
    textAlign: "center",
    marginHorizontal: 15,
    position: "absolute",
    bottom: 60,
    fontWeight: "bold",
  },
});

export default tenFrameGameStyles;
