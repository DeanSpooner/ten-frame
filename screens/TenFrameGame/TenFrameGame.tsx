import { transform } from "@babel/core";
import React, { useState, useEffect } from "react";
import { Text, TouchableOpacity, View } from "react-native";
import Table from "../../components/Table/Table";
import tenFrameGameStyles from "./TenFrameGame.styles";

const TenFrameGame: React.FC<{}> = () => {
  const [counters, setCounters] = useState<number>(0);
  const [score, setScore] = useState<number>(0);
  const [total, setTotal] = useState<number>(0);
  const [correctAnswer, setCorrectAnswer] = useState<boolean>(false);
  const [wrongAnswer, setWrongAnswer] = useState<boolean>(false);

  const lengthOfFrame: number = 10;
  const arrayOfButtons: number[] = Array.from(Array(lengthOfFrame)).map(
    (e, i) => i + 1
  );

  const randomCounters = () => {
    setCounters(Math.floor(Math.random() * 10 + 1));
  };

  const handleAnswer = (num: number) => {
    if (num === counters) {
      setScore(score + 1);
      setTotal(total + 1);
      randomCounters();
      setWrongAnswer(false);
      setCorrectAnswer(true);
      setTimeout(() => {
        setCorrectAnswer(false);
      }, 1000);
    } else {
      setTotal(total + 1);
      setCorrectAnswer(false);
      setWrongAnswer(true);
      setTimeout(() => {
        setWrongAnswer(false);
      }, 1000);
    }
  };

  useEffect(() => {
    randomCounters();
  }, []);

  return (
    <>
      <Text style={tenFrameGameStyles.score}>
        Score: {score} out of {total}
      </Text>
      <Text style={tenFrameGameStyles.text}>How many counters are there?</Text>
      <Table counter={counters} />
      <View style={tenFrameGameStyles.buttonsContainer}>
        {arrayOfButtons.map((num) => (
          <TouchableOpacity
            key={num}
            onPress={() => {
              handleAnswer(num);
            }}
            style={tenFrameGameStyles.container}
          >
            <Text style={tenFrameGameStyles.buttonText}>{num}</Text>
          </TouchableOpacity>
        ))}
      </View>
      {correctAnswer && (
        <Text style={[tenFrameGameStyles.answerMessage, { color: "#009" }]}>
          Correct!
        </Text>
      )}
      {wrongAnswer && (
        <Text style={[tenFrameGameStyles.answerMessage, { color: "#900" }]}>
          Try again!
        </Text>
      )}
    </>
  );
};

export default TenFrameGame;
