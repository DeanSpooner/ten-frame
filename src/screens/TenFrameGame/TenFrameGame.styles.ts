import { Dimensions, StyleSheet } from "react-native";

const tenFrameGameStyles = StyleSheet.create({
  score: {
    color: "#ffe800",
    fontSize: Dimensions.get("window").width / 12,
    marginBottom: 20,
    textAlign: "center",
  },
  time: {
    color: "#00e8ff",
    flex: 1,
    fontSize: Dimensions.get("window").width / 12,
    textAlign: "center",
  },
  timeContainer: {
    justifyContent: "center",
    alignItems: "center",
    marginBottom: 20,
    width: 100,
  },
  buttonsContainer: {
    flexWrap: "wrap",
    flexDirection: "row",
    justifyContent: "center",
  },
  container: {
    backgroundColor: "#fff",
    borderColor: "#000",
    borderWidth: 3,
    borderRadius: 50,
    margin: 5,
    height: Dimensions.get("window").width * 0.17,
    width: "17%",
    justifyContent: "center",
  },
  buttonText: {
    fontSize: 48,
    textAlign: "center",
    justifyContent: "center",
  },
  answerMessage: {
    fontSize: Dimensions.get("window").width / 10,
    textAlign: "center",
    marginHorizontal: 15,
    fontWeight: "bold",
  },
  resetButtonContainer: {
    backgroundColor: "#ff7964",
    border: "#000",
    borderRadius: 500,
    borderWidth: 3,
    height: Dimensions.get("window").width / 2,
    justifyContent: "center",
    width: Dimensions.get("window").width / 2,
  },
  startResetButton: {
    color: "#000",
    fontSize: Dimensions.get("window").width / 12,
    textAlign: "center",
  },
  startButtonContainer: {
    backgroundColor: "#02d1fc",
    border: "#000",
    borderRadius: 500,
    borderWidth: 3,
    height: Dimensions.get("window").width / 2,
    justifyContent: "center",
    width: Dimensions.get("window").width / 2,
  },
});

export default tenFrameGameStyles;
