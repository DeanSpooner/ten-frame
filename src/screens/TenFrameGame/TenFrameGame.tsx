import React, { useState, useEffect } from "react";
import { Text, TouchableOpacity, View } from "react-native";
import Table from "../../components/Table/Table";
import { arrayMaker } from "../../helpers/arrayMaker";
import tenFrameGameStyles from "./TenFrameGame.styles";

const TenFrameGame: React.FC<{}> = () => {
  const [counters, setCounters] = useState<number>(0);
  const [score, setScore] = useState<number>(0);
  const [total, setTotal] = useState<number>(0);
  const [time, setTime] = useState<number>(60);
  const [gameStarted, setGameStarted] = useState<boolean>(false);
  const [correctAnswer, setCorrectAnswer] = useState<boolean>(false);
  const [wrongAnswer, setWrongAnswer] = useState<boolean>(false);

  const lengthOfFrame: number = 10;

  const randomCounters = (): void => {
    setCounters(Math.floor(Math.random() * 10 + 1));
  };

  const handleAnswer = (num: number): void => {
    if (num === counters) {
      setScore(score + 1);
      setTotal(total + 1);
      randomCounters();
      setWrongAnswer(false);
      setCorrectAnswer(true);
      setTimeout(() => {
        setCorrectAnswer(false);
      }, 1000);
    } else {
      setTotal(total + 1);
      setCorrectAnswer(false);
      setWrongAnswer(true);
      setTimeout(() => {
        setWrongAnswer(false);
      }, 1000);
    }
  };

  useEffect(() => {
    randomCounters();
  }, []);

  useEffect(() => {
    if (gameStarted && time > 0) {
      setTimeout(() => {
        setTime(time - 1);
      }, 1000);
    }
    return;
  }, [gameStarted, time]);

  return (
    <>
      {!gameStarted && time !== 0 && (
        <TouchableOpacity
          onPress={() => setGameStarted(true)}
          style={tenFrameGameStyles.startButtonContainer}
        >
          <Text style={tenFrameGameStyles.startResetButton}>Start game!</Text>
        </TouchableOpacity>
      )}
      {gameStarted && time === 0 && (
        <>
          <Text style={tenFrameGameStyles.text}>Great try! You scored </Text>
          <Text style={tenFrameGameStyles.score}>
            {score} out of {total}!
          </Text>
          <TouchableOpacity
            onPress={() => {
              setGameStarted(true);
              setTime(60);
            }}
            style={tenFrameGameStyles.resetButtonContainer}
          >
            <Text style={tenFrameGameStyles.startResetButton}>Try again!</Text>
          </TouchableOpacity>
        </>
      )}
      {gameStarted && time > 0 && (
        <>
          <View style={tenFrameGameStyles.timeContainer}>
            <Text style={tenFrameGameStyles.time}>{time}</Text>
          </View>
          <Text style={tenFrameGameStyles.score}>
            Score: {score} out of {total}
          </Text>
          <Text style={tenFrameGameStyles.text}>
            How many counters are there?
          </Text>
          <Table counter={counters} />
          <View style={tenFrameGameStyles.buttonsContainer}>
            {arrayMaker(lengthOfFrame).map((num) => (
              <TouchableOpacity
                key={num}
                onPress={() => {
                  handleAnswer(num);
                }}
                style={tenFrameGameStyles.container}
              >
                <Text style={tenFrameGameStyles.buttonText}>{num}</Text>
              </TouchableOpacity>
            ))}
          </View>
          {correctAnswer && (
            <Text style={[tenFrameGameStyles.answerMessage, { color: "#009" }]}>
              Correct!
            </Text>
          )}
          {wrongAnswer && (
            <Text style={[tenFrameGameStyles.answerMessage, { color: "#900" }]}>
              Try again!
            </Text>
          )}
        </>
      )}
    </>
  );
};

export default TenFrameGame;
