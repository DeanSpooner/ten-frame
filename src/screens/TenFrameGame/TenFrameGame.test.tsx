import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import TenFrameGame from "./TenFrameGame";

describe("<TenFrameGame />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TenFrameGame />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child", () => {
    const tree = renderer
      .create(<TenFrameGame />)
      .toJSON() as ReactTestRendererJSON[];
    //@ts-ignore
    expect(tree?.children.length).toBe(1);
  });
});
