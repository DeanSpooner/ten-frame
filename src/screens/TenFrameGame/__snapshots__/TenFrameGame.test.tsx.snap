// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TenFrameGame /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 20,
      "minHeight": 1234,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#02d1fc",
        "border": "#000",
        "borderRadius": 500,
        "borderWidth": 3,
        "height": 375,
        "justifyContent": "center",
        "opacity": 1,
        "width": 375,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 62.5,
          "textAlign": "center",
        }
      }
    >
      Start game!
    </Text>
  </View>
</View>
`;
